src/game_battler.cpp

```diff
@@ -437,7 +437,12 @@ int Game_Battler::ApplyConditions() {
 }
 
 void Game_Battler::RemoveBattleStates() {
-	State::RemoveAllBattle(GetStates(), GetPermanentStates());
+	std::vector<bool> removed_states = State::RemoveAllBattle(GetStates(), GetPermanentStates());
+
+	if (!removed_states.empty() && removed_states[0]) {
+		// Death was removed
+		SetHp(1);
+	}
 }
 
 void Game_Battler::RemoveAllStates() {
```

Comment:
Does this also need to set hp to 1 when death is removed?

Reply:
This function is only called as part of FullHeal so doesn't really matter.

But will change it for consistency reasons.